src/subset.c

```diff
@@ -13,13 +13,13 @@ void subsetVectorRaw(SEXP ans, SEXP source, SEXP idx, const bool anyNA)
 
   #define PARLOOP(_NAVAL_)                                        \
   if (anyNA) {                                                    \
-    _Pragma("omp parallel for num_threads(getDTthreads())")       \
+    _Pragma("omp parallel for if (n>=getDTthreads()) num_threads(getDTthreads())")       \
```

Comment:
this means if there's 40 threads, we'll parallelize as soon as we subset 41 items in subset?

Reply:
40 because it is `>=`, not `>`

Reply:
Hmm it seems like a start but I guess there would still be a high overhead vs compute on `n=40` case. May be worth setting up some sort of optimization

Reply:
Yes, that would be, but making that threshold dynamic require a lot of exploration on different hardware and input cases. Current approach is quite straightforward, "if you cannot use all threads, don't bother and do sequentially".

Reply:
Great start.  How about passing `n` into `getDTthreads()` though and doing the `if()` inside `getDTthreads()`?  That way, a parameter could be added inside `getDTthreads()`, ITER_PER_THREAD, so that we and users can more easily experiment.  Some of our parallel regions are clear that, say, 100 or 1000 iterations should happen in single thread before using two threads.  Some other of our parallel regions are already pre-chunked and designed so that each sinlge iteration should be given to each thread. `getDTthreads()` could take a boolean indicating which type that instance is (i.e. whether it is pre-chunked or not).  The user-controlled option would control just the non-pre-chunked parallel regions.

Reply:
Agree, I was unsure about performance implication. I was assuming using `if () num_threads()` will be safer than only `num_threads`. Just checked now and it seems fine to redirect that logic into `num_threads()` (so `getDTthreads`). At least on gcc.
```
if (1>0) num_threads(1) ## use 1th
took 0.188s

if (1>0) num_threads(256) ## use 256 th
took 0.459s

if (1<0) num_threads(256) ## use 1 th
took 0.186s
```
Will move that logic to `getDTthreads` then